torch/csrc/utils/tensor_new.cpp

```diff
@@ -170,7 +170,7 @@ ScalarType infer_scalar_type(PyObject *obj) {
     return ScalarType::Bool;
   }
   if (PyComplex_Check(obj)) {
-    return ScalarType::ComplexDouble;
+    return ScalarType::ComplexFloat;
```

Comment:
I think we should do this instead:

```
switch (torch::tensors::get_default_scalar_type()) {
  case ScalarType::Float: return ScalarType::ComplexFloat;
  case ScalarType::Double: return ScalarType::ComplexDouble;
  default: AT_CHECK(0, "invalid default scalar type for complex");
}
```

Reply:
but doesn't get_default_scalar_type() always return float? Although, I do agree that it would help in ensuring that the behavior for complex tensors is consistent with float, double in case we change the default scalar type to double in future

Reply:
In this case, I'm thinking about the intended use case of `get_default_scalar_type`. The main reason we introduced it was so that we could conveniently bump up the precision of tensors in our test suite, and then get gradcheck working (as double precision is much better for getting gradcheck). Under this motivation, a similar reasoning would apply to complex numbers, since you're also planning to run gradcheck on them. Now, you could add a `get_default_complex_type` but there really shouldn't be any situation where the default scalar type and default complex type should disagree.

Reply:
oh okay makes sense! updated